back-end/src/dms/tests/test_book_endpoints.py

```diff
@@ -359,3 +360,74 @@ def test_delete_books_with_permissions(su_client, test_client, test_user,
                                                        "self"))
     delete_response = test_client.delete(f"/api/dms/books/{book_id}/")
     assert delete_response.status_code == 204
+
+
+@pytest.mark.django_db
+# pylint: disable=too-many-arguments
+def test_delete_book_deletes_cover_and_file(su_client, book_data, author_data,
+                                            publisher_data, subject_data):
+    data = book_data(author_data, publisher_data, subject_data)
+    create_response = send_create_request(su_client, data)
+
+    book = Book.objects.get(pk=create_response.json()["id"])
+    cover_id = book.cover.pk
+    file_id = book.file.pk
+
+    image_path = f"/back-end/media/covers/{create_response.data['cover']}"
+    assert os.path.isfile(image_path)
+
+    file_path = f"/back-end/{create_response.data['file']['content']}"
+    assert os.path.isfile(file_path)
+
+    assert Cover.objects.filter(pk=cover_id).exists()
+    assert File.objects.filter(pk=file_id).exists()
+
+    book_id = create_response.data["id"]
+    su_client.delete(f"/api/dms/books/{book_id}/")
+
+    assert not Cover.objects.filter(pk=cover_id).exists()
+    assert not File.objects.filter(pk=file_id).exists()
+    assert not os.path.isfile(image_path)
+    assert not os.path.isfile(file_path)
+
+
+@pytest.mark.django_db
+# pylint: disable=too-many-arguments
+def test_change_book_deletes_old_cover_and_file(su_client, book_data,
+                                                author_data, publisher_data,
+                                                subject_data, image):
+    data = book_data(author_data, publisher_data, subject_data)
+    create_response = send_create_request(su_client, data)
+
+    book_id = create_response.json()["id"]
+    book = Book.objects.get(pk=book_id)
+    cover_id = book.cover.pk
+    file_id = book.file.pk
+
+    image_path_before = \
+        f"/back-end/media/covers/{create_response.data['cover']}"
+    assert os.path.isfile(image_path_before)
+
+    file_path_before = f"/back-end/{create_response.data['file']['content']}"
+    assert os.path.isfile(file_path_before)
+    assert Cover.objects.filter(pk=cover_id).exists()
+    assert File.objects.filter(pk=file_id).exists()
+
+    file = ContentFile("file_content_new", name="file_new.txt")
+    image_file = image("image_new.png")
+    content = encode_multipart(
+        boundary=BOUNDARY,
+        data={
+            "content": file,
+            "image": image_file,
+        },
+    )
+    update_response = su_client.patch(f"/api/dms/books/{book_id}/",
+                                      data=content,
+                                      content_type=MULTIPART_CONTENT)
+    assert update_response.status_code == 200
+
+    file_path_after = f"/back-end/{update_response.data['file']['content']}"
+    image_path_after = f"/back-end/media/covers/{update_response.data['cover']}"
+    assert image_path_before == image_path_after
```

Comment:
I don't know how to test this correctly. The fact is that when replacing a file, a new one is now loaded into the place of the old one, and before it was put next to it, and a suffix was added to it.


Now I'm checking that the file names match, which means that the old file has been replaced with a new one.

Reply:
Well, is should be possible to copy the old file somewhere before deleting it and compare after it was replaced with the new one.

Reply:
OK, let's do it like this